### PR TITLE
Additonal wxWebView fixes and additions

### PR DESCRIPTION
**Pull request by TcT2k**
* Add fullscreen event (when javascript triggers fullscreen)
* Add find for edge and WKWebView backend
* Fix [#19074](https://trac.wxwidgets.org/ticket/19074)
* Don't trigger erroneous wxEVT_WEBVIEW_ERROR when vetoing wxEVT_WEBVIEW_NAVIGATING  with edge

### CODE REVIEW

File: interface/wx/webview.h [hunk 1]
```diff
@@ -428,6 +428,10 @@ class wxWebViewHandler
     @event{EVT_WEBVIEW_TITLE_CHANGED(id, func)}
        Process a @c wxEVT_WEBVIEW_TITLE_CHANGED event, generated when
        the page title changes. Use GetString to get the title.
+    @event{EVT_WEBVIEW_FULL_SCREEN_CHANGED(id, func)}
+       Process a @c EVT_WEBVIEW_FULL_SCREEN_CHANGED event, generated when
+       the page wants to enter or leave fullscreen. Use GetInt to get the status.
+       Currently only implemented for the edge and WebKit2GTK+ backend.
```
Comment (vadz) [inline interface/wx/webview.h:434]:
```suggestion
       Currently only implemented for the edge and WebKit2GTK+ backend
       and is only available in wxWidgets 3.1.5 or later.
```

File: src/msw/webview_edge.cpp [hunk 2]
```diff
@@ -278,6 +302,22 @@ HRESULT wxWebViewEdgeImpl::OnContentLoading(ICoreWebView2* WXUNUSED(sender), ICo
...
 
+HRESULT wxWebViewEdgeImpl::OnContainsFullScreenElementChanged(ICoreWebView2* WXUNUSED(sender), IUnknown* WXUNUSED(args))
+{
+    BOOL containsFullscreenEvent;
+    HRESULT hr = m_webView->get_ContainsFullScreenElement(&containsFullscreenEvent);
+    if (FAILED(hr))
+        return hr;
+
+    wxWebViewEvent event(wxEVT_WEBVIEW_FULLSCREEN_CHANGED, m_ctrl->GetId(),
+        m_ctrl->GetCurrentURL(), "");
```
Comment (vadz) [inline src/msw/webview_edge.cpp:313]: Nitpick
```suggestion
        m_ctrl->GetCurrentURL(), wxString());
```

**Review by vadz — COMMENTED** [review]
Thanks!

I'll merge this when the CI builds finish (unless you prefer to keep it open if you plan to add more commits to this branch).

**Comment by TcT2k** [comment]
> Thanks!
> 
> I'll merge this when the CI builds finish (unless you prefer to keep it open if you plan to add more commits to this branch).

Don't plan more changes here, should be good to merge, when CI finishes. I applied your suggestions.

**Comment by MaartenBent** [comment]
Based on the diff it looks like you changed all line endings to crlf. I should extend the EOL check with checking for expected EOL.

**Comment by vadz** [comment]
> Based on the diff it looks like you changed all line endings to crlf. I should extend the EOL check with checking for expected EOL.

Yes, I'd like to suggest to use the list of extensions I've posted in a comment before (maybe removing some that we don't use) and checking that EOLs correspond to the expected ones and not just that they're consistent.

**Comment by vadz** [comment]
Thanks again, merging now!

P.S. The [memory leak here](https://github.com/wxWidgets/wxWidgets/pull/2227/checks?check_run_id=1875403286#step:9:59) seems to be real, i.e. it has happened more than once, although I have no idea why does it happen only sometimes (i.e. I wasn't able to reproduce it). This is unrelated to the changes here, of course.

**Pull request merged**